File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.cpp
class Solution {
public:
    string processStr(string s) {
        vector<char>result;
        for (char i : s){
            if (i!='*' && i!='#' && i!='%' ){
                result.push_back(i);
            }else if (i=='*'){
                if (!result.empty()){
                   result.pop_back();
                }
            }else if (i=='#'){
                result.insert(result.end(),result.begin(),result.end());
            }else{
                reverse(result.begin(),result.end());
            }

        }
        string re;

        // if (result.size()==0){
        //     return "";
        // }
        
        for (int i=0;i<result.size();i++){
            re+=result[i];
        }
        return re;
    }
};

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.py
class Solution:
    def processStr(self, s: str) -> str:
        a=[]
        for i in s:
            if i!="*" and i!='#' and i!="%":
                a.append(i)
            elif i=="*":
                if a:
                   a.pop()
            elif i=="#":
                a.extend(a)
            elif i=="%":
                a = a[::-1]
        b=""
        for i in a:
            b+=i


        if len(b)==0:
            return ""
        else:
            return b
            
